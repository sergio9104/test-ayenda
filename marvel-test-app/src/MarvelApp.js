import React, { useEffect, useState } from 'react';
import ComicItem from './components/ComicItem';
import FilterComic from './components/FilterComic';
import Header from './components/Header';
import useGetApi from './hooks/useGetApi';

const MarvelApp = () => {

    const { data, loading } = useGetApi()
    const [comics, setComics] = useState(data);

    useEffect( () => {
        setComics(data);
    }, [data])

    const handleButton = () => {
        setComics(data);
    }

    return(
        <div className='main'>
            <Header />
            <FilterComic setComics={setComics} comics={comics}/>
            <button className='clearBtn' onClick={handleButton}><i class="fa-solid fa-broom"></i> Clear filter</button>
            { loading && <p className="animate__animated animate__flash">Cargando datos...</p> }
            <div className='comics'>
                {
                    comics.map( comic => (
                        <ComicItem
                            key={comic.id}
                            {...comic}
                        />
                    ))
                }
            </div>
        </div>
    )

}

export default MarvelApp;
