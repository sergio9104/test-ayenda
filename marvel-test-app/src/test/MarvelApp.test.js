import { shallow } from "enzyme";
import MarvelApp from "../MarvelApp";

describe('Pruebas en Marvel app', () => {

    test('Debe hacer match con el snapshot', () => {
        const wrapper = shallow(<MarvelApp />)
        expect( wrapper ).toMatchSnapshot();
    })

})