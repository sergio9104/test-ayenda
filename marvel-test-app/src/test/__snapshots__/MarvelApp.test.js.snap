// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Marvel app Debe hacer match con el snapshot 1`] = `
<div
  className="main"
>
  <Header />
  <FilterComic
    comics={Array []}
    setComics={[Function]}
  />
  <button
    className="clearBtn"
    onClick={[Function]}
  >
    <i
      class="fa-solid fa-broom"
    />
     Clear filter
  </button>
  <p
    className="animate__animated animate__flash"
  >
    Cargando datos...
  </p>
  <div
    className="comics"
  />
</div>
`;
